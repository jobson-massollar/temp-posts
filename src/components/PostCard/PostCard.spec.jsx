import { render, screen } from "@testing-library/react"
import { PostCard } from "."

const post = {
    title: 'title 1',
    id: 1,
    body: 'body 1',
    cover: 'img/img.png'
}

describe('<PostCard />', () => {

    it('should render', () => {
        // const { debug } = render(<PostCard post={post} />)
        // debug();  // Show the code generated by the render

        render(<PostCard post={post} />)

        const img = screen.getByRole('img')

        expect(img).toHaveAttribute('src', post.cover)
        expect(img).toHaveAttribute('alt', post.title)

        expect(screen.getByText(post.title + ' ' + post.id)).toBeInTheDocument(); // Title paragraph
        expect(screen.getByText(post.body)).toBeInTheDocument(); // Body paragraph
    })

    it('should match snapshot', () => {
        const { container } = render(<PostCard post={post} />)
        
        expect(container.firstChild).toMatchSnapshot()
    })
})
