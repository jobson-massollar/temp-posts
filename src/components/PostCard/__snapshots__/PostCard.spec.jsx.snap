// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <p
    class="post-title"
  >
    title 1
     
    1
  </p>
  <p
    class="post-body"
  >
    body 1
  </p>
</div>
`;
